aptos-move/e2e-move-tests/src/tests/gas.rs

```diff
@@ -18,43 +47,68 @@ fn test_gas() {
     let account_2_address = *account_2.address();
     let account_3_address = *account_3.address();
 
-    print_gas_cost(
+    let profile_gas = std::env::var("PROFILE_GAS").is_ok();
+
+    let run: fn(&mut MoveHarness, &str, &Account, TransactionPayload) = if !profile_gas {
+        |harness, function, account, payload| {
+            print_gas_cost(function, harness.evaluate_gas(account, payload));
+        }
+    } else {
+        |harness, function, account, payload| {
+            let (log, gas_used) = harness.evaluate_gas_with_profiler(account, payload);
+            save_profiling_results(function, &log);
+            print_gas_cost(function, gas_used);
+        }
+    };
+
+    let publish: fn(&mut MoveHarness, &str, &Account, &Path) = if !profile_gas {
+        |harness, name, account, path| {
+            print_gas_cost(name, harness.evaluate_publish_gas(account, path));
+        }
+    } else {
+        |harness, name, account, path| {
+            let (log, gas_used) = harness.evaluate_publish_gas_with_profiler(account, path);
+            save_profiling_results(name, &log);
+            print_gas_cost(name, gas_used);
+        }
+    };
```

Comment:
@msmouse made the change you requested.

I also switched the logic, making the gas profile activated by default. This will allow us to get some decent code coverage for the gas profiler itself.

Reply:
not exactly what I meant, but okay.. 😂